### PR TITLE
Feat: [M3-6805] AGLB Primary Nav Entry

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR adds a primary nav entry for the AGLB feature
- Not visible to customers (behind a feature flag)
- Customized beta chip (see self review comments)
- Will be iterated on as needed

## Preview 📷

Screenshot 2023-07-13 at 5 15 12 PM

## How to test 🧪
1. Pull code locally, enable both MSW & the AGLB flag
2. Confirm presence and behavior of new menu entry
3. Verify there's no regression with the nav when the flag is off

### CODE REVIEW

File: packages/manager/src/components/PrimaryNav/PrimaryNav.tsx [hunk 1]
```diff
@@ -59,6 +60,7 @@ interface PrimaryLink {
   onClick?: (e: React.ChangeEvent<any>) => void;
   hide?: boolean;
   isBeta?: boolean;
+  betaChipClassName?: string;
```
Comment (abailly-akamai) [inline packages/manager/src/components/PrimaryNav/PrimaryNav.tsx:63]: This new Props helps customizing the positioning of the beta chip cause in this case, our menu item will span three lines with the beta chip. I wanted to extend the width of the primary nav but that's something to thread lightly with, so in the meantime we'll just do this. Also, none of this is customer facing at the moment so we can iterate.

File: packages/manager/src/components/PrimaryNav/PrimaryNav.styles.ts [hunk 2]
```diff
@@ -71,9 +71,7 @@ const useStyles = makeStyles<void, 'linkItem'>()(
       display: 'flex',
       alignItems: 'center',
       cursor: 'pointer',
-      height: 36,
-      lineHeight: 0,
-      padding: '12px 16px',
+      padding: '8px 16px',
```
Comment (abailly-akamai) [inline packages/manager/src/components/PrimaryNav/PrimaryNav.styles.ts:74]: this preserve the exact same spacing as we have currently, without being restrictive about two lines menu items

**Review by cpathipa — APPROVED** [review]

**Comment by jdamore-linode** [comment]
@abailly-akamai Added integration tests for this. They test that the menu entry is present and functional when AGLB is enabled, and is absent when AGLB is not enabled.

To run the new tests:

```bash
yarn cy:run -s "cypress/e2e/core/loadBalancers/load-balancer-navigation.spec.ts"
```

**Review by jdamore-linode — APPROVED** [review]

**Comment by abailly-akamai** [comment]
@jdamore-linode great thank you! it's probably more robust than my unit test entry

**Pull request merged**